src/nvim/ex_cmds.c

```diff
@@ -1408,32 +1408,30 @@ char_u *make_filter_cmd(char_u *cmd, char_u *itmp, char_u *otmp)
   }
 
   if (itmp != NULL) {
-    strncat(buf, " < ", len);
-    strncat(buf, (char *) itmp, len);
+    strncat(buf, " < ", len - 1);
+    strncat(buf, (const char *)itmp, len - 1);
```

Comment:
This can still overrun the buffer since the second `strncat()` isn't taking into account the size of the write in the first `strncat()`.  The second call should be `strncat(buf, (const char *)itmp, len - 4);` (or we could implement `xstrlcat`).